design/ceph/ceph-nfs-ganesha.md

```diff
@@ -213,8 +226,125 @@ The following directories should **not** be shared between SSSD and other contai
+Default PrincipalName is "nfs".
+Default for keytab path is reportedly empty. Rook can use `/etc/ker
+
+Create a new RADOS object named `kerberos` to configure Kerberos.
+```ini
+NFS_KRB5
+{
+   PrincipalName = nfs ; # <-- set from spec.security.kerberos.principalName
+   KeytabPath = /etc/krb5.keytab ;
+   Active_krb5 = YES ;
+}
+```
+
+Add the following line to to the config object (`conf-nfs.${nfs-name}`) to reference the new
+`kerberos` RADOS object. Remove this line from the config object if Kerberos is disabled.
+```
+%url "rados://.nfs/${nfs-name}/kerberos"
+```
+
+These steps can be done from the Rook operator.
+
+#### Kerberos config
+Below is the `/etc/krb5.conf` default from the ceph container:
+```ini
+# To opt out of the system crypto-policies configuration of krb5, remove the
+# symlink at /etc/krb5.conf.d/crypto-policies which will not be recreated.
+includedir /etc/krb5.conf.d/
+
+[logging]
+    default = FILE:/var/log/krb5libs.log
+    kdc = FILE:/var/log/krb5kdc.log
+    admin_server = FILE:/var/log/kadmind.log
+
+[libdefaults]
+    dns_lookup_realm = false
+    ticket_lifetime = 24h
+    renew_lifetime = 7d
+    forwardable = true
+    rdns = false
+    pkinit_anchors = FILE:/etc/pki/tls/certs/ca-bundle.crt
+    spake_preauth_groups = edwards25519
+#    default_realm = EXAMPLE.COM
+    default_ccache_name = KEYRING:persistent:%{uid}
+
+[realms]
+# EXAMPLE.COM = {
+#     kdc = kerberos.example.com
+#     admin_server = kerberos.example.com
+# }
+
+[domain_realm]
+# .example.com = EXAMPLE.COM
+# example.com = EXAMPLE.COM
+```
+
+Rook should set its own krb.conf like so. Rook can create `krb5.conf` in an EmptyDir from an init
+container and mount it to the `nfs-ganesha` container in the same fashion as for `nsswitch.conf`.
+Inline comments explain changes made compared to the default file.
+```ini
+includedir /etc/krb5.conf.d/    # assume we want to keep crypto-policies config
+includedir /etc/krb5.conf.rook/ # load configFiles content from users
+
+[logging]
+    default = STDERR # only log to stderr by default
+
+# remove all other defaults (namely 'libdefaults') so that users have full flexibility to set what they want.
+```
+
+**QUESTION:** Do we want to keep this `crypto-policies` file? Do we think users might want to remove
+it? Will overriding `[libdefaults] permitted_enctypes` be sufficient? Will they have to name their
+file `zz-*` to ensure it overwrites `crypto-policies`?
+```
+$# cat /etc/krb5.conf.d/crypto-policies
+[libdefaults]
+permitted_enctypes = aes256-cts-hmac-sha1-96 aes256-cts-hmac-sha384-192 camellia256-cts-cmac aes128-cts-hmac-sha1-96 aes128-cts-hmac-sha256-128 camellia128-cts-cmac
+```
+
+#### Enabling kerberos for an export
+**QUESTION:**
+`sectype` enables Kerberos for an export and is set on the Export definition. I don't see that
+option in `ceph nfs export ...` commands. Do we need to manually add `sectype` to all exports?
+How will CSI do this? Should be part of Ceph API, no?
```

Comment:
It should get part of the Ceph-API, yes. Possibly other options need to get included there as well.

Reply:
Is this something you could start working on? I wonder if it would be good to have the ability to add generic export options rather than continuously adding flags to the Ceph CLI API when we find something is missing.